docs/zh_CN/extension/source.md

```diff
@@ -1,83 +1,83 @@
+源的主要任务是实现_open_方法，且应该和创建到外部系统的连接保持同步。然后从外部系统连续接收数据，并将接收到的消息发送到消费通道。消费通道接受SourceTuple接口，该接口由消息正文的映射和可选元数据的另一个映射组成。通常，使用`api.NewDefaultSourceTuple(message, meta)`命令创建SourceTuple。元数据可以是任何值得记录的内容。例如，消息的合格主题。第一个参数是StreamContext指针。您可以从中检索上下文信息和日志等。它也是go上下文的实现，因此您可以监听Done（）通道以了解父流是否已退出。对于在连接或接收过程中发生的任何错误，请使用此方法进行处理。如果错误无法处理，请将其发送到errCh。默认情况下，如果从errCh收到任何错误，则该规则将终止。
 
 ```go
 //Should be sync function for normal case. The container will run it in go func
 Open(ctx StreamContext, consumer chan<- SourceTuple, errCh chan<- error)
-```  
+```
 
-The last method to implement is _Close_ which literally close the connection. It is called when the stream is about to terminate. You could also do any clean up work in this function.
+最后要实现的方法是_Close_，它实际上用来关闭连接。 当流即将终止时调用它。 您也可以在此功能中执行任何清理工作。
 
 ```go
 Close(ctx StreamContext) error
 ```
 
-As the source itself is a plugin, it must be in the main package. Given the source struct name is mySource. At last of the file, the source must be exported as a symbol as below. There are [2 types of exported symbol supported](overview.md#plugin-development). For source extension, states are usually needed, so it is recommended to export a constructor function.
+由于源本身是一个插件，因此它必须位于主程序包中。 给定源结构名称为mySource。 在文件的最后，必须将源作为符号导出，如下所示。 有 [2种类型的导出符号](overview.md#plugin-development)。 对于源扩展，通常需要状态，因此建议导出构造函数。
 
 ```go
 function MySource() api.Source{
     return &mySource{}
 }
 ```
 
-The [Randome Source](../../../plugins/sources/random.go) is a good example.
+[Randome Source](../../../plugins/sources/random.go) 是一个很好的示例。
+
+### 处理配置
+
+Kuiper配置的格式为yaml，它提供了一个集中位置 _/etc_ 来保存所有配置。 在其中，为源配置提供了一个子文件夹 _sources_，同时也适用于扩展源。
+
+Kuiper扩展支持配置系统自动读取yaml文件中的配置，并将其输入到源的_Configure_方法中。 如果在流中指定了[CONF_KEY](../streams.md#create-stream) 属性，则将输入该键的配置。 否则，将使用默认配置。
 
-### Deal with configuration
+要在源中使用配置，必须遵循以下约定：
+ 1. 您的配置文件名称必须与_.so_文件相同，并且必须 采用驼峰式命名法， 首字母大写。 例如，MySource.yaml。
+  2. yaml文件必须位于_etc/sources_内。
+  3. 可以在 [此处](../rules/sources/mqtt.md)找到yaml文件的格式。
```

Comment:
A configuration system is supported for Kuiper extension which will automatically read the configuration in yaml file and feed into the _Configure_ method of the source. If the [CONF_KEY](../streams.md#create-stream) property is specified in the stream, the configuration of that key will be fed. Otherwise, the default configuration is used.

上面这部分没有翻译。